### PR TITLE
test(api): add unit tests for JobSettingsScheduler

### LINKED ISSUE #79: Missing test coverage for ImapSyncScheduler
## Bug

`ImapSyncScheduler` has no unit tests. The following methods are untested:

- `ExecuteAsync` — startup loading of existing mailboxes
- `ScheduleMailboxAsync` — scheduling with immediate trigger
- `RescheduleMailboxAsync` — delete + re-create job
- `UnscheduleMailboxAsync` — job removal

These are critical orchestration methods that coordinate Quartz.NET job lifecycle with mailbox CRUD operations.

### Acceptance Criteria

- [ ] Test that `ExecuteAsync` schedules jobs for all existing mailboxes
- [ ] Test that `ScheduleMailboxAsync` with `triggerImmediately: true` fires the job
- [ ] Test that `RescheduleMailboxAsync` updates the interval
- [ ] Test that `UnscheduleMailboxAsync` removes the job

### Found in

PR #74 code review

### PR DESCRIPTION
**Pull request by mamu7211**
## Summary
- Adds 9 unit tests for `JobSettingsScheduler`, the `ManagedJob`-based replacement for `ImapSyncScheduler`
- Covers `ExecuteAsync` (startup scheduling), `ScheduleJobAsync`, `UnscheduleJobAsync`, and `RescheduleJobAsync`
- Includes positive paths, edge cases (unregistered job types, non-existent jobs), and operation ordering

Closes #79

## Test plan
- [x] All 9 new tests pass locally
- [ ] CI passes

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/Feirb.Api.Tests/Services/JobSettingsSchedulerTests.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
+using Feirb.Api.Data;
+using Feirb.Api.Data.Entities;
+using Feirb.Api.Services;
+using FluentAssertions;
```
[the remark this answers is not in the document]
Comment (mamu7211) [reply tests/Feirb.Api.Tests/Services/JobSettingsSchedulerTests.cs]: Fixed in c22c69f -- removed unused FluentAssertions import.
